File: cpp_intro2algo/test_queue.cpp
#include <iostream>
#include "dataStructures.h"

using namespace std;


int main(){

    int a[10];
    int pop;
    Queue qa;
    Queue qb(a, 8);

    for (int i=0; i<10; i++){
        a[i] = i;
    }

    cout << "An empty queue" << endl;
    cout << "Empty = " << boolalpha << qa.is_empty() << " ";
    cout << "Head = " << qa.get_head() << endl;
    qa.showq();
    qa.enqueue(1);
    qa.showq();
    qa.dequeue();
    qa.showq();

    cout << "An random queue:" << endl;
    cout << "Empty = " << boolalpha << qb.is_empty() << " ";
    cout << "Head = " << qb.get_head() << " ";
    cout << "Tail = " << qb.get_tail() << endl;
    qb.showq();
    cout << "Push in: 1" << endl;
    qb.enqueue(1);
    qb.showq();
    pop = qb.dequeue();
    cout << "Pop out: " << pop << endl;
    qb.showq();
    cout << "Push in: " << pop << endl;
    qb.enqueue(pop);
    qb.showq();

    return 0;
}

File: py_intro2algo/test_queue.py
from __future__ import print_function
from dataStructures import Queue

a = []
b = range(10)
qa = Queue(a)
qb = Queue(b)

print("An empty queue: ", qa)
print(type(qa))
print("Empty = ", qa.is_empty(), "Head = ",  qa.get_head())
qa.showq()

print("Push in: a")
qa.enqueue('a')
qa.showq()

pop = qa.dequeue()
print("Pop out: ", pop)
qa.showq()


print("An randon queue: ", qb)
print("Empty = ", qb.is_empty(), "Head = ", qb.get_head(), "Tail = ", qb.get_tail())
qb.showq()

print("Push in: a")
qb.enqueue('a')
qb.showq()

pop = qb.dequeue()
print("Pop out: ", pop)
qb.showq()

print("Push in:", pop)
qb.enqueue(pop)
qb.showq()
